### ISSUE TITLE
Expanded controlled state is based on sorted row data

### ISSUE THREAD

**Issue by ldstein** [react-tools/react-table#483]
## What version of React-Table are you using?
Your bug may already be fixed in the latest release. Run `yarn upgrade react-table`!
6.5.1

## What bug are you experiencing, or what feature are you proposing?
If you expand a row while maintaining a expanded control state, then sort the data, the wrong row will be expanded.

Perhaps the question really is was expanded row state intentionally tied to a row index or should it really map to the actual data which is populating a row?

## What are the steps to reproduce the issue?
1. Visit https://react-table.js.org/#/story/controlled-component
2. Expand the first row
3. Click once on 'Last name' column header
4. The first row is expanded (different data to what you previously expanded)

**Comment by tannerlinsley** [comment]
It is intentionally tied to the index of the view layer. Not the data layer. It is possible to change this. 

This is a question of implementation. We invite you to join the slack channel and ask for help there or on stack overflow. Thanks!

**Comment by dowi** [comment]
This is something I also ran into. 
Is there a way to know/calculate  the final index of an "input data row"? 
Thanks a lot

**Comment by diranl** [comment]
Running into the same issue, can you provide more details?

**Comment by diranl** [comment]
For future reference, found the solution here: https://github.com/react-tools/react-table/issues/547#issuecomment-336737630

Change from `index` to `viewIndex` in order to refer to the correct row in the view:

```
private expandRow = (row) => {
    const { expanded } = this.state;
    expanded[row.viewIndex] = !expanded[row.viewIndex];
    this.setState({ expanded });
  };
```

**Comment by Atirinsaf** [comment]
Hi there,
I have used React-table 7 datagrid with few customizations , where I have hidden all headers and just using a single column as a narrow hierarchy component. I have implemented server side sorting on this datagrid(narrow hierarchy component) by clicking on a button (This button is in a component from where I am calling react-table datagrid) . On initial render, I can see all icons in a collapsed format, when i click on this expand icon, children level API call is getting made and children data got appended post api call response. When I click on sort button, sorting order changes (eg:- "ASC","DESC") , then I see rows position which I have opened in previous sort order , those positions are showing expandable icon as expanded rather then collapsed. Can somebody help me here???
